statics/src/test/java/integration/ShadowDomInsideIFrameTest.java

```diff
@@ -0,0 +1,39 @@
+package integration;
+
+import com.codeborne.selenide.Configuration;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static com.codeborne.selenide.Condition.text;
+import static com.codeborne.selenide.Condition.value;
+import static com.codeborne.selenide.Selectors.shadowCss;
+import static com.codeborne.selenide.Selenide.$;
+import static com.codeborne.selenide.Selenide.switchTo;
+
+class ShadowDomInsideIFrameTest extends IntegrationTest {
+  @BeforeEach
+  void setUp() {
+    openFile("page_with_shadow_dom_inside_iframe.html");
+  }
+
+  @Test
+  void getTargetElementViaShadowHostInsideIFrame() {
+    switchTo().frame("iframe_page");
+    $(shadowCss("p", "#shadow-host"))
+      .shouldHave(text("Inside Shadow-DOM"));
+  }
+
+  @Test
+  void setValueViaShadowDomInsideIFrame() {
```

Comment:
add assume not FF check to avoid test fail on FF

Reply:
thanks! done